Neos.Flow/Classes/Core/Booting/Scripts.php

```diff
@@ -736,9 +741,73 @@ public static function buildPhpCommand(array $settings)
             $command .= ' -c ' . escapeshellarg($useIniFile);
         }
 
+        self::ensureWebSubrequestsUseCurrentlyRunningPhpVersion($command);
+
         return $command;
     }
 
+    /**
+     * Compares the realpath of the configured PHP binary (if any) with the one flow was called with in a CLI request.
+     * This avoids config errors where users forget to set Neos.Flow.core.phpBinaryPathAndFilename in CLI.
+     *
+     * @param string phpBinaryPathAndFilename
+     * @throws Exception
+     */
+    protected static function ensureCLISubrequestsUseCurrentlyRunningPhpBinary($phpBinaryPathAndFilename)
+    {
+        // Do nothing for non-CLI requests
+        if (PHP_SAPI !== 'cli') {
+            return;
+        }
+
+        // Resolve any symlinks that the configured php might be pointing to
+        $configuredPhpBinaryPathAndFilename = realpath($phpBinaryPathAndFilename);
+
+        // if the configured PHP binary is empty here, the file does not exist. We ignore that here because it is checked later in the script.
+        if (strlen($configuredPhpBinaryPathAndFilename) === 0) {
+            return;
+        }
+
+        if (strcmp(PHP_BINARY, $configuredPhpBinaryPathAndFilename) !== 0) {
```

Comment:
Two questions here:

1. Why not directly always using `PHP_BINARY` for the sub process instead of relying on some config. AFAIK this also works reliably on Windows
1. Isn't `PHP_BINARY === $configuredPhpBinaryPathAndFilename` easier to grasp than using `strcmp`?

Reply:
From what I remember the PHP_BINARY was unreliable on shared environments with multiple PHP versions or something like that? I don't really remember the case.

Reply:
PHP_BINARY is not reliable on Windows from Web. It will just point to the process that launched php (e.g. apache/nginx).

Reply:
`PHP_BINARY` is *never* to rely on for web requests. It **is** reliable on cli and always points to the path of the currently in use PHP binary. This whole code is [only executed on cli](https://github.com/neos/flow-development-collection/pull/1391/files#diff-3dae3262c03fe101b952964ee70e2ddbR759). So why not just rely on `PHP_BINARY`, because it would just work.

Like that: 

```php
if (PHP_SAPI === 'cli') {
   return PHP_BINARY
}

return $settings['configuredPHPBinaryForWebRequests'];
```

This works reliably in practice across operating systems for me for years on a fairly large user base.

The only complaint I received was from a user that called the script with additional options for the php binary like so: `php -derror_reporting=0`, which is not easily solvable, except such options are configurable as well.

Reply:
But this method is not responsible for finding the php binary, but rather check if the found (configured) php binary matches the one that this command is currently run with to detect configuration errors :) Also, we probably shouldn't use PHP_BINARY in the one case (CLI) and rely on a configuration setting (`phpBinaryPathAndFilename`) in the other (Web), even if that would make it work in at least one case reliably.